internal/api/observability/output_types.go

```diff
@@ -0,0 +1,119 @@
+package observability
+
+import (
+	"fmt"
+	log "github.com/ViaQ/logerr/v2/log/static"
+	obsv1 "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/set"
+	"os"
+)
+
+func OutputTypeUnknown(t obsv1.OutputType) error {
+	return fmt.Errorf("Unknown output type %q", t)
+}
+
+type Outputs []obsv1.OutputSpec
+
+// ConfigmapNames returns a unique set of unordered configmap names
+func (outputs Outputs) ConfigmapNames() []string {
+	names := set.New[string]()
+	for _, o := range outputs {
+		if o.TLS != nil {
+			names.Insert(ConfigmapsForTLS(o.TLS.TLSSpec)...)
+		}
+	}
+	return names.UnsortedList()
+}
+
+// SecretNames returns a unique set of unordered secret names
+func (outputs Outputs) SecretNames() []string {
+	secrets := set.New[string]()
+	for _, o := range outputs {
+		if o.TLS != nil {
+			secrets.Insert(SecretsForTLS(o.TLS.TLSSpec)...)
```

Comment:
Should we `continue` here to next iteration?

Reply:
Not the way I read it.   We still need any other secrets that are non-TLS, correct?